device/core/src/client.ts

```diff
@@ -571,7 +323,11 @@ export class Client extends EventEmitter {
     /*Codes_SRS_NODE_DEVICE_CLIENT_16_016: [The ‘complete’ method shall throw a ReferenceError if the ‘message’ parameter is falsy.] */
     if (!message) throw new ReferenceError('message is \'' + message + '\'');
 
-    this._fsm.handle('complete', message, completeCallback);
+    this._transport.complete(message, (err, result) => {
+      if (completeCallback) {
```

Comment:
>completeCallback [](start = 10, length = 16)

Is it possible that the transport doesn't support completion?  If so, should we throw in this method and the following few methods?

Reply:
the fact is that all transports have these methods, and it's the transport (MQTT in that case) that ends up throwing. it's a little different for other features (where some methods are missing from transports) but I haven't had time to unify that. Maybe later :)

---
In reply to: [145239756](https://github.com/Azure/azure-iot-sdk-node/pull/109#discussion_r145239756) [](ancestors = 145239756)